include/ginkgo/core/matrix/csr.hpp

```diff
@@ -792,6 +793,10 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
 
     std::unique_ptr<Diagonal<ValueType>> extract_diagonal() const override;
 
+    std::unique_ptr<Csr<ValueType, IndexType>> create_submatrix(
```

Comment:
I think this could also use some documentation?
```suggestion
    std::unique_ptr<Csr> create_submatrix(
```